go/arrow/cdata/cdata_testframework.c

```diff
@@ -0,0 +1,88 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I don't understand why this file exists in addition to `cdata_fulltest.c`.
Overall, the file naming here is a bit mysterious. Why is there a "test framework" module in addition to the actual test module, for example? Is it a Go convention?

Reply:
So there's two aspects here:

1. You can't actually include "cgo" compilations in test modules with Go currently. So there needs to be cdata_test_framework.go in order to expose the functions in the .c files to be called by the actual `_test.go` file as part of the unit tests. 
2. There is both a cdata_fulltest.c and a cdata_testframework.c file because of the way I developed the tests. I first put something together that I thought would be a framework for the tests (`cdata_testframework.c`) before following your suggestion and also expanding the test coverage which is when I created `cdata_fulltest.c`. There's honestly no reason to keep this as a separate file instead of just incorporating it into the cdata_fulltest.c file. Basically, I called it `cdata_test_framework.c` solely because of the limitation on having cgo in test files with the idea that I was creating a framework before pivoting to be more concrete tests. 

I'll merge the contents of this into the `cdata_fulltest.c` file just for consistency since there's no need for two separate .c files.

Reply:
updated with this change.